### PR TITLE
20557-Update-Zinc-and-Zodiac

### PR DESCRIPTION
**Pull request by svenvc**
https://pharo.manuscript.com/f/cases/20557/Update-Zinc-and-Zodiac

Sync Pharo 7 with content of upstream Zn+Zdc repositories
 http://mc.stfx.eu/Zodiac
 http://mc.stfx.eu/ZincHTTPComponents

Versions
 Zinc-Character-Encoding-Core-SvenVanCaekenberghe.64
 Zinc-Character-Encoding-Tests-SvenVanCaekenberghe.43
 Zinc-FileSystem-SvenVanCaekenberghe.15
 Zinc-HTTP-SvenVanCaekenberghe.474
 Zinc-Resource-Meta-Core-SvenVanCaekenberghe.67
 Zinc-Resource-Meta-FileSystem-SvenVanCaekenberghe.9
 Zinc-Tests-SvenVanCaekenberghe.245
 Zinc-Zodiac-SvenVanCaekenberghe.44
 Zodiac-Core-TorstenBergmann.48
 Zodiac-Tests-SvenVanCaekenberghe.22

### CODE REVIEW

**Comment by svenvc** [comment]
Wow it passed, super !

**Review by astares — APPROVED** [review]
Looks good (after manual review on GH).

**Comment by svenvc** [comment]
Thank you all.

File: src/Zinc-Character-Encoding-Core/ZnCrPortableWriteStream.class.st [hunk 1]
```diff
@@ -0,0 +1,61 @@
+"
+I am a write stream wrapping a second stream. Whenever they ask me to write a cr, a lf, or a crlf I'll instead print a portable new line depending on the platform I'm on.
+
+stream := '' writeStream.
+converter := ZnCrPortableWriteStream on: stream.
+converter cr; cr; lf; nextPut: $a.
+stream contents
+"
+Class {
+	#name : #ZnCrPortableWriteStream,
```
Comment (jecisc) [inline src/Zinc-Character-Encoding-Core/ZnCrPortableWriteStream.class.st:10]: Wasn't this rename ZnNewlinePortableWritestream?

**Comment by svenvc** [comment]
Yes, but I can't remove this one in Zn proper as it would break older pharo versions when loading Zn (it is used during logging by the system), it should be deprecated though.

Anyway, in Pharo 7 dev it could be removed.

**Pull request merged**